north/src/runtime/state.rs

```diff
@@ -320,13 +312,88 @@ impl State {
         }
     }
 
+    async fn is_installed(&self, npk: &Path) -> Result<bool, InstallFailure> {
+        let container_id = extract_manifest(npk.into(), &self.signing_keys)?.name;
+        Ok(self.applications.get(&container_id).is_some())
+    }
+
     /// Install a npk from give path
-    // pub async fn install(&mut self, npk: &Path) -> result::Result<(), Error> {
-    //     npk::install(self, npk)
-    //         .await
-    //         .map_err(Error::InstallationError)?;
-    //     Ok(())
-    // }
+    pub async fn install(
+        &mut self,
+        npk: &Path,
+        msg_id: MessageId,
+        container_dir: Option<std::path::PathBuf>,
+        install_file_name: String,
+        tx: sync::Sender<Message>,
+    ) {
+        info!(
+            "try to install {}, checking the installed apps:",
+            install_file_name,
+        );
+        for app in self.applications() {
+            info!("- {}", app);
```

Comment:
This log is not good. We have other logs in eg logcat and they can interleave. Put everything on one line or add some context to " - {}"

Reply:
yes...was a debug output during development that needs to be removed